pkg/controller/jobframework/webhook/defaulter.go

```diff
@@ -58,3 +75,34 @@ func (h *losslessDefaulter) Handle(ctx context.Context, req admission.Request) a
 	}
 	return response
 }
+
+// getUnknownPaths returns a list of unknown paths.
+func (h *losslessDefaulter) getUnknownPaths(raw []byte) (sets.Set[string], error) {
+	obj := h.object.DeepCopyObject()
+	strictErrors, err := json.UnmarshalStrict(raw, obj, json.DisallowUnknownFields)
+	if err != nil {
+		return nil, err
+	}
+	paths := sets.New[string]()
+	for _, err := range strictErrors {
+		if fieldError, ok := err.(json.FieldError); ok {
```

Comment:
Could we get other errors as well? Maybe we could trigger them to test that scenario? If we cannot trigger them easily this is fine.

Reply:
As I can see [here](https://github.com/kubernetes-sigs/json/blob/bc3834ca7abd3a90f03ef00a27ad80cb892f9c21/json.go#L87) it should implement `FieldError` interface.

Reply:
what happens if the go type is defined with an integer field, but the input object has a string?
Would that fail here or a different place?
I guess, in general, in that case we should fail the whole request.

Reply:
It should fail on `response := h.Handler.Handle(ctx, req)`.

Reply:
Added separate test case to check it.